score.rb

```diff
@@ -17,37 +17,65 @@ def initialize(frames)
 
   attr_accessor :frames
 
+  def single_frame_score(frame)
+    return frame.rolls.sum if frame.tenth_frame?
 
-  def single_frame_score(current_frame)
-    return current_frame.rolls.sum if current_frame.tenth_frame?
+    score_frame_type.each do |frame_type, score_frame|
+      return score_frame.call(frame) if frame_type?(frame_type, frame)
+    end
+  end
 
-    throws = current_frame.rolls
+  # TODO: is this method name descriptive enough?
+  # Method checks if the frame type to score matches the frame type in score_frame_type
+  def frame_type?(frame_type, frame)
+    send(frame_type, frame.rolls)
+  end
 
-    case
-    when strike?(throws.first) then score_strike(current_frame)
-    when spare?(throws) then score_spare(current_frame)
-    when open?(throws) then throws.sum
-    end
+  def score_frame_type
+    # Methods wrapped in Proc
+    {
+      strike?: Proc.new { |frame| score_strike(frame) },
+      spare?: Proc.new { |frame| score_spare(frame) } ,
+      open?: Proc.new { |frame| score_open(frame) }
+    }
```

Comment:
I find the Proc to be clear in terms of which methods are being called and that frame is passed to the respective method. It feels like possibly a lot of overhead? 

I am still reading up on use cases and when to/not to use it.

Note to self: correct the arrow spacing on lines 44-45

Reply:
Ruled out for use here, I think. 

A regular proc is flexible with arguments (if I understand correctly). I suspect we gain nothing here with that added laxity and we potentially obfuscate issues arising from too many/few arguments.